File: parse.py
import sqlite3
import datetime
import urllib3
import zipfile
import csv

http = urllib3.PoolManager()
response = http.request('GET', 'https://ga-covid19.ondemand.sas.com/docs/ga_covid_data.zip', preload_content=False)
zip_path = '/tmp/ga_covid_data.zip'
chunk_size = 4
with open(zip_path, 'wb') as out:
    while True:
        data = response.read(chunk_size)
        if not data:
            break
        out.write(data)

directory_path = '/tmp/ga_covid_data/'
with zipfile.ZipFile(zip_path, 'r') as zip_ref:
    zip_ref.extractall(directory_path)

csv_file_name = 'countycases.csv'
csv_path = directory_path + csv_file_name
with open(zip_path) as csvfile:
    reader = csv.reader(csvfile, delimiter=',')
    connection = sqlite3.connect('caseupdates.sqlite')
    cursor = connection.cursor()
    cursor.execute('select max(set_id) from CASES')
    set_id = cursor.fetchone()[0] + 1
    for row in reader:
        if row[0] != 'county_resident': # skip first line
            county = row[0]
            cases = row[1]
            deaths = row[2]
            cursor.execute(
                'INSERT INTO CASES (DATETIME, COUNTY, CASES, DEATHS, SET_ID) VALUES (?, ?, ?, ?, ?)',
                (str(datetime.datetime.now()), county, int(cases), int(deaths), int(set_id))
            )
    connection.commit()
    connection.close()

File: parse.js
const puppeteer = require('puppeteer');

(async () => {
  try {
    const browser = await puppeteer.launch({
      // headless: false
    });
    const page = await browser.newPage();
    await page.goto('https://ga-covid19.ondemand.sas.com/');
    await page.waitForSelector(
        '#react-tabs-5 > div > div > table > tbody > ' +
        'tr:nth-child(1) > td.MuiTableCell-root.MuiTableCell-body.MuiTableCell-' +
        'alignLeft.MuiTableCell-sizeSmall', {
          visible: true,
        });
    const html = await page.content();
    const myregexp = /<td class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeSmall" style="border: none;">([a-zA-Z ]+)<\/td><td class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall" style="border: none;">(\d+)<\/td><td class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall" style="border: none;">[.\d-]+<\/td><td class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall" style="border: none;">(\d+)/g;
    let match = myregexp.exec(html);
    const db = require('better-sqlite3')('caseupdates.sqlite');
    const row = await db.prepare('select max(set_id) as setId from CASES').get();
    const setId = row.setId + 1;
    const insert = db.prepare('INSERT INTO CASES (DATETIME, COUNTY, CASES, DEATHS, SET_ID) VALUES (?, ?, ?, ?, ?)');
    while (match != null) {
      const d = new Date();
      const datestring = (d.getFullYear()) + "-" + ("0"+(d.getMonth()+1)).slice(-2) + "-"
          + ("0" + d.getDate()).slice(-2) + " " + ("0" + d.getHours()).slice(-2) + ":" + ("0" + d.getMinutes()).slice(-2)
          + ":" + ("0" + d.getSeconds()).slice(-2) + ".000000";
      insert.run(datestring, `${match[1]}`, `${match[2]}`, `${match[3]}`, setId);
      match = myregexp.exec(html);
    }
    await db.close();
    await browser.close();
  } catch (error) {
    console.log(error);
  }
})();
